### PR TITLE
Add new tnf command to check the test results

### PR DESCRIPTION
**Pull request by jmontesi**
The command compares the test results of a CNFCERT execution using the log file with those provided in a YAML template to check that they match. If not, a table showing the expected vs actual results will be printed on stdout.

To generate the test results YAML template from a log file:

* `tnf check results --generate-template`

To check the test results against a reference template:

* `tnf check results --template <expected_results.yaml> --log-file <cnf-certsuite.log>`

### CODE REVIEW

File: cmd/tnf/check/results/results.go [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
```
Comment (rdavid) [inline cmd/tnf/check/results/results.go:15]: This text is GPL. The project is Apache-2.0 license. I think following line is enough:
`Copyright (C) 2020-2024 Red Hat, Inc.`

I know we have these lines in a lot of files, needs to clean up.

File: cmd/tnf/check/results/results.go [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+	for _, testCase := range expectedTestResultsList.Fail {
+		expectedTestResults[testCase] = resultFail
+	}
+
+	return expectedTestResults, nil
+}
+
+func printTestResultsMismatch(mismatchedTestCases []string, actualResults, expectedResults map[string]string) {
+	fmt.Printf("\n")
+	fmt.Println(strings.Repeat("-", 96)) //nolint:gomnd // table line
```
Comment (edcdavid) [inline cmd/tnf/check/results/results.go:163]: optional: there are libraries automating text tables in go. For instance, the one used in tasty: https://github.com/syohex/go-texttable

**Review by edcdavid — APPROVED** [review]

File: cmd/tnf/check/results/results.go (hunk 1 above)
Comment (jmontesi) [reply cmd/tnf/check/results/results.go]: Done.

File: cmd/tnf/check/results/results.go (hunk 2 above)
Comment (jmontesi) [reply cmd/tnf/check/results/results.go:149]: Thanks for the suggestion. But I think it's better not to have this dependency for the limited use of tables we have in our project.

**Review by greyerof — APPROVED** [review]

File: cmd/tnf/check/results/results.go [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+	fmt.Printf("| %-58s %-19s %s |\n", "TEST_CASE", "EXPECTED_RESULT", "ACTUAL_RESULT")
+	fmt.Println(strings.Repeat("-", 96)) //nolint:gomnd // table line
+	for _, testCase := range mismatchedTestCases {
+		expectedResult, exist := expectedResults[testCase]
+		if !exist {
+			expectedResult = resultMiss
+		}
+		actualResult, exist := actualResults[testCase]
+		if !exist {
+			actualResult = resultMiss
```
[the remark this answers is not in the document]
Comment (jmontesi) [reply cmd/tnf/check/results/results.go:159]: `mismatchedTestCases` was also populated using the `expectedTestResults` (see line 75). So, this situation would happen if there were test cases defined in the template that did not appear in the log file.

[the remark this answers is not in the document]
Comment (greyerof) [reply cmd/tnf/check/results/results.go:159]: Understood. Thanks!

**Pull request merged**